File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        sum_ = 0
        mp = {0: -1}
        
        for i in range(n):
            sum_ += nums[i]
            remainder = sum_ % k

            if remainder in mp:
                if i - mp[remainder] >= 2:
                    return True
            else:
                mp[remainder] = i

        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;
        int sum = 0;
        mp.put(0, -1);
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            int remainder = sum % k;
            if (mp.containsKey(remainder)) {
                if (i - mp.get(remainder) >= 2) {
                    return true;
                }
            } else {
                mp.put(remainder, i);
            }
        }
        return false;
    }
}